Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 9417
"GET /static/images/slide1.jpg HTTP/1.1" 304 0
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /authentication/login/ HTTP/1.1" 200 3838
"GET /static/images/logo.webp HTTP/1.1" 304 0
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 22979
"GET /technician/farm-locations/ HTTP/1.1" 200 15142
"GET /technician/profile/ HTTP/1.1" 200 19121
Forbidden (CSRF token from POST incorrect.): /authentication/login/
"POST /authentication/login/ HTTP/1.1" 403 2503
Forbidden (CSRF token from POST incorrect.): /authentication/login/
"POST /authentication/login/ HTTP/1.1" 403 2503
"GET /authentication/login/ HTTP/1.1" 200 3859
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /farmer/dashboard/ HTTP/1.1" 200 6963
C:\Users\user\Desktop\soils-sense\apps\administrator\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\authentication\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\authentication\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\soilsense\settings.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\soilsense\settings.py changed, reloading.
Watching for file changes with StatReloader
"GET /farmer/analytics/ HTTP/1.1" 200 12664
"GET /farmer/profile/ HTTP/1.1" 200 15469
"GET /authentication/reset/ HTTP/1.1" 200 3254
"GET /farmer/soil-data/ HTTP/1.1" 200 27584
"GET /farmer/dashboard/ HTTP/1.1" 200 6963
"GET /farmer/soil-data/ HTTP/1.1" 200 27584
"GET /farmer/profile/ HTTP/1.1" 200 15469
"GET /farmer/analytics/ HTTP/1.1" 200 12664
"GET /farmer/recommendations/ HTTP/1.1" 200 15736
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 22979
"GET /technician/dashboard/ HTTP/1.1" 200 22979
"GET /technician/dashboard/ HTTP/1.1" 200 22979
Not Found: /favicon.ico
"GET /favicon.ico HTTP/1.1" 404 3911
"GET /technician/reports/ HTTP/1.1" 200 13306
"GET /technician/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 22979
"GET /technician/dashboard/ HTTP/1.1" 200 22979
"GET /technician/dashboard/ HTTP/1.1" 200 22979
Not Found: /favicon.ico
"GET /favicon.ico HTTP/1.1" 404 3911
"GET /technician/settings/ HTTP/1.1" 200 14599
"GET /technician/analytics/ HTTP/1.1" 200 16581
"GET /technician/settings/ HTTP/1.1" 200 14599
"GET /technician/profile/ HTTP/1.1" 200 19121
"GET /technician/sensor-config/ HTTP/1.1" 200 16623
"GET /technician/soil-readings/ HTTP/1.1" 200 11572
"GET /technician/dashboard/ HTTP/1.1" 200 22979
"GET /technician/soil-readings/ HTTP/1.1" 200 11572
"GET /technician/profile/ HTTP/1.1" 200 19121
"GET /technician/settings/ HTTP/1.1" 200 14599
"GET /technician/profile/ HTTP/1.1" 200 19121
"GET /technician/settings/ HTTP/1.1" 200 14599
C:\Users\user\Desktop\soils-sense\apps\technician\forms.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/settings/ HTTP/1.1" 200 15196
Not Found: /media/users/profile_images/default.webp
"GET /media/users/profile_images/default.webp HTTP/1.1" 404 4235
"GET /technician/settings/ HTTP/1.1" 200 15297
Not Found: /media/users/profile_images/default.webp
"GET /media/users/profile_images/default.webp HTTP/1.1" 404 4235
"GET /technician/profile/ HTTP/1.1" 200 19121
Internal Server Error: /technician/settings/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\smartif.py", line 180, in translate_token
    op = OPERATORS[token]
         ~~~~~~~~~^^^^^^^
KeyError: "user.image.url.endswith:'default.webp'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 316, in settings
    return render(request, 'technician/settings.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 299, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 961, in do_if
    condition = TemplateIfParser(parser, bits).parse()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 894, in __init__
    super().__init__(*args, **kwargs)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\smartif.py", line 171, in __init__
    mapped_tokens.append(self.translate_token(token))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\smartif.py", line 182, in translate_token
    return self.create_var(token)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 897, in create_var
    return TemplateLiteral(self.template_parser.compile_filter(value), value)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 609, in compile_filter
    return FilterExpression(token, self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 710, in __init__
    raise TemplateSyntaxError(
django.template.exceptions.TemplateSyntaxError: Could not parse the remainder: ':'default.webp'' from 'user.image.url.endswith:'default.webp''
"GET /technician/settings/ HTTP/1.1" 500 285302
Internal Server Error: /technician/settings/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\smartif.py", line 180, in translate_token
    op = OPERATORS[token]
         ~~~~~~~~~^^^^^^^
KeyError: "user.image.url.endswith:'default.webp'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 316, in settings
    return render(request, 'technician/settings.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 299, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 961, in do_if
    condition = TemplateIfParser(parser, bits).parse()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 894, in __init__
    super().__init__(*args, **kwargs)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\smartif.py", line 171, in __init__
    mapped_tokens.append(self.translate_token(token))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\smartif.py", line 182, in translate_token
    return self.create_var(token)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 897, in create_var
    return TemplateLiteral(self.template_parser.compile_filter(value), value)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 609, in compile_filter
    return FilterExpression(token, self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 710, in __init__
    raise TemplateSyntaxError(
django.template.exceptions.TemplateSyntaxError: Could not parse the remainder: ':'default.webp'' from 'user.image.url.endswith:'default.webp''
"GET /technician/settings/ HTTP/1.1" 500 285302
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/settings/ HTTP/1.1" 200 15431
Internal Server Error: /technician/settings/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 320, in settings
    return render(request, 'technician/settings.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 299, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 489, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 487, in parse
    filter_expression = self.compile_filter(token.contents)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 609, in compile_filter
    return FilterExpression(token, self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 705, in __init__
    filter_func = parser.find_filter(filter_name)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 615, in find_filter
    raise TemplateSyntaxError("Invalid filter: '%s'" % filter_name)
django.template.exceptions.TemplateSyntaxError: Invalid filter: 'add_class'
"GET /technician/settings/ HTTP/1.1" 500 238727
"GET /technician/settings/ HTTP/1.1" 200 16335
"GET /technician/profile/ HTTP/1.1" 200 19121
"GET /technician/analytics/ HTTP/1.1" 200 16581
"GET /technician/sensor-config/ HTTP/1.1" 200 16623
"GET /technician/dashboard/ HTTP/1.1" 200 22979
"GET /technician/farm-locations/ HTTP/1.1" 200 15142
"GET /technician/settings/ HTTP/1.1" 200 17247
"POST /technician/settings/ HTTP/1.1" 302 0
"GET /technician/settings/ HTTP/1.1" 200 16943
"GET /media/users/profile_images/bruno.png HTTP/1.1" 200 312527
"GET /technician/profile/ HTTP/1.1" 200 19612
Internal Server Error: /technician/profile/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\smartif.py", line 180, in translate_token
    op = OPERATORS[token]
         ~~~~~~~~~^^^^^^^
KeyError: "user.image.url.endswith:'default.webp'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 149, in profile
    return render(request, 'technician/profile.html')
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 299, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 961, in do_if
    condition = TemplateIfParser(parser, bits).parse()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 894, in __init__
    super().__init__(*args, **kwargs)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\smartif.py", line 171, in __init__
    mapped_tokens.append(self.translate_token(token))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\smartif.py", line 182, in translate_token
    return self.create_var(token)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 897, in create_var
    return TemplateLiteral(self.template_parser.compile_filter(value), value)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 609, in compile_filter
    return FilterExpression(token, self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 710, in __init__
    raise TemplateSyntaxError(
django.template.exceptions.TemplateSyntaxError: Could not parse the remainder: ':'default.webp'' from 'user.image.url.endswith:'default.webp''
"GET /technician/profile/ HTTP/1.1" 500 241033
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/profile/ HTTP/1.1" 200 19435
"GET /media/users/profile_images/bruno.png HTTP/1.1" 304 0
"GET /technician/settings/ HTTP/1.1" 200 16943
"GET /technician/profile/ HTTP/1.1" 200 19435
"GET /technician/settings/ HTTP/1.1" 200 16943
"GET /media/users/profile_images/bruno.png HTTP/1.1" 304 0
